### ISSUE TITLE
November 8th, 2020 - Storage Deal Issues

### ISSUE THREAD

**Issue by jimmylee** [filecoin-project/slate#573]
This is a snippet from a report about how Slate failed in certain aspects (overall success) in 2020. The purpose of this issue is for PL to reference the Filecoin storage deal issues we encountered that caused us to stop making storage deals completely through Textile's infrastructure.

## (3) — "Slate has a very low success rate of Filecoin storage deals."

This is the unfortunate event that spurred the initial disdain for Slate:WWW. I acknowledge Slate:WWW is a dead-end for people who want to use the Filecoin Network or even re-use our example code to learn how to develop on Filecoin. The problem is summarized in this excerpt:

> The end result is that, even for our most advanced users like Textile, storage deal success rates are <25%. And this is aggregated across hundreds of users. This is **untenable**. For individual users, the story is even worse. Textile has had to tell Slate:WWW to stop making Filecoin storage deals because the volume of deals, given the unreliability of deals, means that deals get stuck in the pipeline for days/weeks. 

Slate:WWW's infrastructure is capable of making 1200+ storage deals for unique real buckets with real data within an hour. Here is an example of the workers we run in parallel:

<img width="1792" alt="Screen Shot 2020-09-30 at 11 17 45 PM" src="https://user-images.githubusercontent.com/310223/98460407-06833480-2159-11eb-84d4-d232a8cf9661.png">

What went wrong:

* No one has a directory of reliable miners that can seal storage deal data and efficient rates. Textile is not an exception to this. The closest thing we have is this cross referenced list I produced: https://sentinel.slate.host/api/mapped-static-global-miners
* Textile found out that encrypted bucket storage deals do not work. Half of our deals are encrypted on behalf of the Slate:WWW users to respect privacy.
* Some miners are asking for too much FIL.
* Data transfer problems: https://github.com/filecoin-project/lotus/issues/4755 (thanks @whyrusleeping)
* Deals that were listed as successful in https://slate.host/api/v1/deals occasionally had miners that were no longer available. @ribasushi graciously provided miners to help the original Slingshot V1 contest get started but had to take them down. We need a method to repair miners: https://github.com/textileio/textile/issues/389

Action plan:

* We will see if [Marcus's Equinix data center](https://www.equinix.com/) miners can work reliably. I am going to shadow this work and learn as much as I can.
* I am working with @whyrusleeping to build a real miner so I understand how to make storage deals like the other experts we have.
* I will implement my own storage deal broker service so I understand the full stack of issues that involve multiple storage deals. Other people currently use Amazon S3 as a hot storage mechanism. I'll try something locally.
* I will work as closely as possible with @jsign and Textile to be as helpful as I can be.

**Comment by rvagg** [comment]
Where's the code that generated https://sentinel.slate.host/api/mapped-static-global-miners? I'm particularly interested in how `"minDealDuration"` was derived since it's not part of the QueryAsk data.

**Comment by jimmylee** [comment]
You can find the source here! https://github.com/filecoin-project/slate-sentinel/blob/master/pages/api/mapped-static-global-miners.js

Most likely cross referenced from here: https://indexes.pow.buidllabs.textile.io/index/ask

We're about to make a ton of deals again against some North American miners, you can see the storage config option being used here: https://github.com/filecoin-project/slate/blob/main/scripts/worker-heavy-stones.js
